src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Emitter.cs

```diff
@@ -86,7 +86,7 @@ namespace {lc.Namespace}
                 // loop until you find top level nested class
                 while (parent != null)
                 {
-                    parentClasses.Add($"partial {parent.Keyword} {parent.Name} {parent.Constraints}");
```

Comment:
>Constraints [](http://example.com/codeflow?start=87&length=11)

Wondering, does removing the constraints is the right solution here? wouldn't that somehow can cause other problems? why we didn't just included "using" namespace instead?

Reply:
Refer to comment https://github.com/dotnet/runtime/issues/58550#issuecomment-1010203525. It's the cleanest and simplest fix.

Reply:
Do we care if someone used this generated class with other types which not originally intended by the constraints?  Or is this an unlikely or not supported scenario?

Reply:
the class itself would still have the constraints as specified by the user. Since this is a partial declaration, in this diff we are removing the constraints only from the generated part.